lib/Dialect/D2M/Transforms/LowerToLayout.cpp

```diff
@@ -272,23 +287,52 @@ class D2MSplitCompoundLayoutRewriter : public OpRewritePattern<ToLayoutOp> {
   public:
     explicit BounceTypeBuilder(MLIRContext *ctx) : ctx(ctx) {}
 
+    // Computes a workable bounce shape grid for a virtual grid.
+    llvm::SmallVector<int64_t>
+    computeVirtualGridBounceShape(ArrayRef<int64_t> virtualGridShape,
+                                  ArrayRef<int64_t> deviceGridShape) const {
+
+      // TODO: Generalize to N dimensions.
```

Comment:
:warning: **google\-readability\-todo** :warning:
missing username/bug in TODO
```suggestion
      // TODO(nsmith): Generalize to N dimensions.
```

Reply:
Fixed.